File: DheerajKumar/problem32.cpp
//32. WAP to print the reverse of a number given by user.

#include<iostream>
using namespace std;

int reverse(int number){
    if(number <= 0){
        return -1;
    }
    
    int lastDigit = 0, reverseNumber = 0;
    while(number > 0){
        lastDigit = number % 10;
        reverseNumber = (reverseNumber * 10) + lastDigit;
        number = number/10; 
    }

    return reverseNumber;
}

int main(){
    int number;
    cout<<"Enter the number -> ";
    cin>> number;

    cout<<"The reverse number = "<< reverse(number);
}

File: LovelyKumari/BasicProgramming/Problem32.java
import java.util.Scanner;

public class Problem32 {
    // WAP to print the reverse of a number given by user
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = scanner.nextInt();

        int reverse = 0 ;
        while (num != 0) {
            int digit = num % 10;     
            reverse = reverse * 10 + digit;  
            num /= 10;  
        }
        System.out.println("The reversed number is: "+ reverse);
    }

    
}
